frontends/web/src/components/confirm/Confirm.tsx

```diff
@@ -14,85 +14,74 @@
-        this.setState({
+    confirmation = (message: string, cb: TCallback, customButtonText?: string) => {
+        callback.current = cb;
+        setState({
             active: true,
             message,
             customButtonText,
         });
-    }
+    };
 
-    private respond = (input: boolean) => {
-        this.callback(input);
-        this.setState({
+    const respond = (response: boolean) => () => {
+        callback.current(response);
+        setState({
             active: false,
         });
-    }
-
-    private decline = () => {
-        this.respond(false);
-    }
+    };
 
-    private accept = () => {
-        this.respond(true);
+    const { message, active, customButtonText } = state;
+    if (!active) {
+        return null;
     }
-
-    public render() {
-        const { t } = this.props;
-        const { message, active, customButtonText } = this.state;
-        return active ? (
-            <Dialog title={t('dialog.confirmTitle')} onClose={this.decline}>
-                <div className="columnsContainer half">
-                    <div className="columns">
-                        <div className="column">
-                            {
-                                message ? message.split('\n').map((line, i) => (
-                                    <p
-                                        key={i}
-                                        className={ i === 0 ? 'first' : '' }>
-                                        <SimpleMarkup tagName="span" markup={line} />
-                                    </p>
-                                )) : null
-                            }
-                        </div>
-                    </div>
+    return <Dialog title={t('dialog.confirmTitle')} onClose={respond(false)}>
+        <div className="columnsContainer half">
+            <div className="columns">
+                <div className="column">
+                    {
+                        message ? message.split('\n').map((line, i) => (
+                            <p
+                                key={i}
+                                className={i === 0 ? 'first' : ''}>
+                                <SimpleMarkup tagName="span" markup={line} />
+                            </p>
+                        )) : null
+                    }
                 </div>
-                <DialogButtons>
-                    <Button primary onClick={this.accept}>{customButtonText ? customButtonText : t('dialog.confirm')}</Button>
-                    <Button transparent onClick={this.decline}>{t('dialog.cancel')}</Button>
-                </DialogButtons>
-            </Dialog>
-        ) : null;
-    }
-}
-
-const TranslatedConfirm = translate()(Confirm);
-
-export { confirmation };
-export { TranslatedConfirm as Confirm };
+            </div>
+        </div>
+        <DialogButtons>
+            <Button primary onClick={respond(true)}>{customButtonText ? customButtonText : t('dialog.confirm')}</Button>
```

Comment:
Why does this work? `respond(true)` is of type `()`. It looks like it would be executed immediately on rendering, but onClick should take a function of type `() => void`, e.g. `() => respond(true)`?

Reply:
It's syntax sugar called curring, under some circumstances it would be slightly more performant.

I also had to read twice, see my comment: 
https://github.com/digitalbitbox/bitbox-wallet-app/pull/1612#discussion_r791885941

I'll update the PR and remove it.

Reply:
i see, i mistook the function => arrow as a type declaration arrow :see_no_evil: 

it would have helped to add a type to `respond`.